cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/sourcecontrol/SourceControlMetadataMigrationService.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.sourcecontrol;
+
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.google.inject.Inject;
+import io.cdap.cdap.app.store.Store;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.internal.app.store.AppMetadataStore;
+import io.cdap.cdap.internal.app.store.NamespaceSourceControlMetadataStore;
+import io.cdap.cdap.proto.id.ApplicationReference;
+import io.cdap.cdap.proto.sourcecontrol.SourceControlMeta;
+import io.cdap.cdap.sourcecontrol.RepositoryCleanupService;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service for migrating source control metadata from app_spec table to source control table.
+ * Starting from 6.10.1 we have introduced a separate table for storing scm metadata. As this
+ * information was previously stored in app_spec table we need to run this migration
+ */
+public class SourceControlMetadataMigrationService extends AbstractScheduledService {
```

Comment:
Do we really need to run it on schedule?

Can it be a `AbstractIdleService` instead?

Reply:
We can run it only once per service start. The only downside is if the thread stops because of any uncaught exception (maybe DB is unreachable) we need to wait till next restart for the data to be populated. 
We also do not want to block on this service to be complete running for the service to start as SCM data is anyway should be eventually consistent
@tivv Do you have any suggestion here ?

Reply:
Modified it so that in case success it stops itself. Otherwise retries.
@tivv @itsankit-google Please check and let me know if this makes sense

Reply:
SGTM, will wait for @tivv's approval.